pkg/agent/openflow/packetin.go

```diff
@@ -46,17 +52,29 @@ func (c *client) StartPacketInHandler(stopCh <-chan struct{}) {
 	if len(c.packetInHandlers) == 0 {
 		return
 	}
+	// Subscribe packetin for TraceFlow with reason 1
 	ch := make(chan *ofctrl.PacketIn)
 	err := c.SubscribePacketIn(uint8(ofprAction), ch)
 	if err != nil {
 		klog.Errorf("Subscribe PacketIn failed %+v", err)
 		return
 	}
 	packetInQueue := workqueue.NewNamed("packetIn")
-	go c.parsePacketIn(packetInQueue, stopCh)
+	go c.parsePacketIn(packetInQueue, stopCh, tfPacketHandlerName)
+
+	// Subscribe packetin for CNP Logging with reason 2
+	chNP := make(chan *ofctrl.PacketIn)
```

Comment:
Is chNP used somewhere?
Pass reason as an argument in to function StartPacketInHandler and call separately for traceflow and CNP from cmd/antrea-agent/agent.go?

Reply:
chNP is only used here. It might be possible that both traceflow and CNP are enabled, so we need separate channels for them.
Another way is to use one channel for receiving all the packets, then go though all the existing packetin handlers to check which one could process it.
Wondering about your opinion on this. Thanks!

Reply:
My question was why is the chNP channel is not used anywhere else in the function unlike ch, which is used to receive packets and add into the queue.

Separate channels sound good to me as they correspond to two different reasons. However, this function may need some refactoring to handle pktInHandler with any given reason.

Reply:
Yes, thanks for the comment. I fixed the bug in a recent commit to receive packet from chNP.
Due to finding that reason corresponds to actual ovs reason, this function will be further changed. Thanks!